libs/jwst-codec/src/doc/document.rs

```diff
@@ -107,7 +107,7 @@ pub struct Doc {
     opts: DocOptions,
 
     pub(crate) store: StoreRef,
-    pub(crate) publisher: Arc<DocPublisher>,
+    pub publisher: Arc<DocPublisher>,
```

Comment:
Does it need to be public? `doc.sub()` is not enough?

Reply:
used to prevent the update observer when we migrating or load a workspace from disk :https://github.com/toeverything/OctoBase/blob/61eca9019abe45b45e54dd7cbcb0d56c98127c2d/libs/jwst-storage/src/storage/docs/utils.rs#L8-L15